{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from dammit.fileio.gff3 import GFF3Parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.x.pfam-A.gff3\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/base.py:79: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  return pd.concat(self, ignore_index=True)\n"
     ]
    }
   ],
   "source": [
    "annotations = GFF3Parser(filename=gff_file).read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Dbxref', 'Name', 'Note']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_file = names.dropna(axis=0,how='all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45659</th>\n",
       "      <td>Transcript_10001</td>\n",
       "      <td>\"Pfam:PF00085.16\"</td>\n",
       "      <td>Thioredoxin</td>\n",
       "      <td>Thioredoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45660</th>\n",
       "      <td>Transcript_10002</td>\n",
       "      <td>\"Pfam:PF07910.9\"</td>\n",
       "      <td>Peptidase_C78</td>\n",
       "      <td>Peptidase family C78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45667</th>\n",
       "      <td>Transcript_10009</td>\n",
       "      <td>\"Pfam:PF13637.2\"</td>\n",
       "      <td>Ank_4</td>\n",
       "      <td>Ankyrin repeats (many copies)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45683</th>\n",
       "      <td>Transcript_10011</td>\n",
       "      <td>\"Pfam:PF10250.5\"</td>\n",
       "      <td>O-FucT</td>\n",
       "      <td>GDP-fucose protein O-fucosyltransferase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45684</th>\n",
       "      <td>Transcript_10012</td>\n",
       "      <td>\"Pfam:PF05724.7\"</td>\n",
       "      <td>TPMT</td>\n",
       "      <td>Thiopurine S-methyltransferase (TPMT)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  seqid             Dbxref           Name  \\\n",
       "45659  Transcript_10001  \"Pfam:PF00085.16\"    Thioredoxin   \n",
       "45660  Transcript_10002   \"Pfam:PF07910.9\"  Peptidase_C78   \n",
       "45667  Transcript_10009   \"Pfam:PF13637.2\"          Ank_4   \n",
       "45683  Transcript_10011   \"Pfam:PF10250.5\"         O-FucT   \n",
       "45684  Transcript_10012   \"Pfam:PF05724.7\"           TPMT   \n",
       "\n",
       "                                          Note  \n",
       "45659                              Thioredoxin  \n",
       "45660                     Peptidase family C78  \n",
       "45667            Ankyrin repeats (many copies)  \n",
       "45683  GDP-fucose protein O-fucosyltransferase  \n",
       "45684    Thiopurine S-methyltransferase (TPMT)  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_file.to_csv(\"pfam.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OrthoDB results "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.x.OrthoDB.best.gff3\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    }
   ],
   "source": [
    "annotations = GFF3Parser(filename=gff_file).read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name']]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_file = names.dropna(axis=0,how='all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_1</td>\n",
       "      <td>PPA10102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>ENSFALP00000015835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>G1X5W6_ARTOA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_10001</td>\n",
       "      <td>C1GUP1_PARBA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              seqid                Name\n",
       "0      Transcript_1            PPA10102\n",
       "1    Transcript_100        G4UTF3_NEUT9\n",
       "2   Transcript_1000  ENSFALP00000015835\n",
       "3  Transcript_10000        G1X5W6_ARTOA\n",
       "4  Transcript_10001        C1GUP1_PARBA"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>protein_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_1</td>\n",
       "      <td>PPA10102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>ENSFALP00000015835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>G1X5W6_ARTOA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_10001</td>\n",
       "      <td>C1GUP1_PARBA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              seqid          protein_id\n",
       "0      Transcript_1            PPA10102\n",
       "1    Transcript_100        G4UTF3_NEUT9\n",
       "2   Transcript_1000  ENSFALP00000015835\n",
       "3  Transcript_10000        G1X5W6_ARTOA\n",
       "4  Transcript_10001        C1GUP1_PARBA"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_file.columns = [\"seqid\", \"protein_id\"]\n",
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "ortho_names = pd.read_csv(\"./../ODB8_EukOGs_genes_ALL_levels.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>odb8_level</th>\n",
       "      <th>odb8_og_id</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>odb8_prot_id</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG800004</td>\n",
       "      <td>H6BPB0_EXODN</td>\n",
       "      <td>858893:000f19</td>\n",
       "      <td>Exophiala dermatitidis NIH/UT8656</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG80000C</td>\n",
       "      <td>G0W340_NAUDC</td>\n",
       "      <td>1071378:000955</td>\n",
       "      <td>Naumovozyma dairenensis CBS 421</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG80000V</td>\n",
       "      <td>M3BSB9_SPHMS</td>\n",
       "      <td>692275:001da6</td>\n",
       "      <td>Sphaerulina musiva SO2202</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG800010</td>\n",
       "      <td>E2LV46_MONPE</td>\n",
       "      <td>554373:002928</td>\n",
       "      <td>Moniliophthora perniciosa FA553</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG80001M</td>\n",
       "      <td>E9DCK5_COCPS</td>\n",
       "      <td>443226:00127d</td>\n",
       "      <td>Coccidioides posadasii str. Silveira</td>\n",
       "      <td>Ornithine aminotransferase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   odb8_level odb8_og_id    protein_id    odb8_prot_id  \\\n",
       "0  4751:Fungi  EOG800004  H6BPB0_EXODN   858893:000f19   \n",
       "1  4751:Fungi  EOG80000C  G0W340_NAUDC  1071378:000955   \n",
       "2  4751:Fungi  EOG80000V  M3BSB9_SPHMS   692275:001da6   \n",
       "3  4751:Fungi  EOG800010  E2LV46_MONPE   554373:002928   \n",
       "4  4751:Fungi  EOG80001M  E9DCK5_COCPS   443226:00127d   \n",
       "\n",
       "                               organism         uniprot_description  \n",
       "0     Exophiala dermatitidis NIH/UT8656                         NaN  \n",
       "1       Naumovozyma dairenensis CBS 421                         NaN  \n",
       "2             Sphaerulina musiva SO2202                         NaN  \n",
       "3       Moniliophthora perniciosa FA553                         NaN  \n",
       "4  Coccidioides posadasii str. Silveira  Ornithine aminotransferase  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_names.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_w_names = pd.merge(new_file, ortho_names, on = \"protein_id\", how = \"inner\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>odb8_level</th>\n",
       "      <th>odb8_og_id</th>\n",
       "      <th>odb8_prot_id</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_1</td>\n",
       "      <td>PPA10102</td>\n",
       "      <td>33208:Metazoa</td>\n",
       "      <td>EOG83FJ79</td>\n",
       "      <td>54126:00270f</td>\n",
       "      <td>Pristionchus pacificus</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_11383</td>\n",
       "      <td>PPA10102</td>\n",
       "      <td>33208:Metazoa</td>\n",
       "      <td>EOG83FJ79</td>\n",
       "      <td>54126:00270f</td>\n",
       "      <td>Pristionchus pacificus</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_30709</td>\n",
       "      <td>PPA10102</td>\n",
       "      <td>33208:Metazoa</td>\n",
       "      <td>EOG83FJ79</td>\n",
       "      <td>54126:00270f</td>\n",
       "      <td>Pristionchus pacificus</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "      <td>4751:Fungi</td>\n",
       "      <td>EOG8WWQ3V</td>\n",
       "      <td>510952:0016ff</td>\n",
       "      <td>Neurospora tetrasperma FGSC 2509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "      <td>4890:Ascomycota</td>\n",
       "      <td>EOG8MPH73</td>\n",
       "      <td>510952:0016ff</td>\n",
       "      <td>Neurospora tetrasperma FGSC 2509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              seqid    protein_id       odb8_level odb8_og_id   odb8_prot_id  \\\n",
       "0      Transcript_1      PPA10102    33208:Metazoa  EOG83FJ79   54126:00270f   \n",
       "1  Transcript_11383      PPA10102    33208:Metazoa  EOG83FJ79   54126:00270f   \n",
       "2  Transcript_30709      PPA10102    33208:Metazoa  EOG83FJ79   54126:00270f   \n",
       "3    Transcript_100  G4UTF3_NEUT9       4751:Fungi  EOG8WWQ3V  510952:0016ff   \n",
       "4    Transcript_100  G4UTF3_NEUT9  4890:Ascomycota  EOG8MPH73  510952:0016ff   \n",
       "\n",
       "                           organism uniprot_description  \n",
       "0            Pristionchus pacificus                 NaN  \n",
       "1            Pristionchus pacificus                 NaN  \n",
       "2            Pristionchus pacificus                 NaN  \n",
       "3  Neurospora tetrasperma FGSC 2509                 NaN  \n",
       "4  Neurospora tetrasperma FGSC 2509                 NaN  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_w_names.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "93453"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ortho_w_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21729"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_w_names.drop_duplicates(subset='seqid', keep='first',  inplace=True)\n",
    "len(ortho_w_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21780"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_names = pd.read_csv(\"./../ODB8_EukOGs_genes_ALL_levels.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "OG_names = pd.read_csv(\"./../odb9v1_OG2genes.tab\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EOG09000000</th>\n",
       "      <th>106582:001e18</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EOG09000000</td>\n",
       "      <td>303518:003f23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EOG09000000</td>\n",
       "      <td>32507:0040ef</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EOG09000000</td>\n",
       "      <td>8153:00339b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EOG09000001</td>\n",
       "      <td>106582:004341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EOG09000001</td>\n",
       "      <td>303518:003afd</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   EOG09000000  106582:001e18\n",
       "0  EOG09000000  303518:003f23\n",
       "1  EOG09000000   32507:0040ef\n",
       "2  EOG09000000    8153:00339b\n",
       "3  EOG09000001  106582:004341\n",
       "4  EOG09000001  303518:003afd"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "OG_names.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1000565:000000</th>\n",
       "      <th>1000565</th>\n",
       "      <th>1000565:000000.1</th>\n",
       "      <th>F5R740</th>\n",
       "      <th>METUNv1_00006</th>\n",
       "      <th>\\N</th>\n",
       "      <th>Transposase of ISThsp18, IS1182 family</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000565:000001</td>\n",
       "      <td>1000565</td>\n",
       "      <td>1000565:000001</td>\n",
       "      <td>F5R741</td>\n",
       "      <td>METUNv1_00011</td>\n",
       "      <td>\\N</td>\n",
       "      <td>Transposase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000565:000002</td>\n",
       "      <td>1000565</td>\n",
       "      <td>1000565:000002</td>\n",
       "      <td>F5R742</td>\n",
       "      <td>METUNv1_00018</td>\n",
       "      <td>\\N</td>\n",
       "      <td>hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000565:000003</td>\n",
       "      <td>1000565</td>\n",
       "      <td>1000565:000003</td>\n",
       "      <td>F5R743</td>\n",
       "      <td>METUNv1_00019</td>\n",
       "      <td>\\N</td>\n",
       "      <td>Transposase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000565:000004</td>\n",
       "      <td>1000565</td>\n",
       "      <td>1000565:000004</td>\n",
       "      <td>F5R744</td>\n",
       "      <td>METUNv1_00035</td>\n",
       "      <td>\\N</td>\n",
       "      <td>PPIase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000565:000005</td>\n",
       "      <td>1000565</td>\n",
       "      <td>1000565:000005</td>\n",
       "      <td>F5R745</td>\n",
       "      <td>METUNv1_00036</td>\n",
       "      <td>\\N</td>\n",
       "      <td>Endopeptidase Clp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   1000565:000000  1000565 1000565:000000.1  F5R740  METUNv1_00006  \\N  \\\n",
       "0  1000565:000001  1000565   1000565:000001  F5R741  METUNv1_00011  \\N   \n",
       "1  1000565:000002  1000565   1000565:000002  F5R742  METUNv1_00018  \\N   \n",
       "2  1000565:000003  1000565   1000565:000003  F5R743  METUNv1_00019  \\N   \n",
       "3  1000565:000004  1000565   1000565:000004  F5R744  METUNv1_00035  \\N   \n",
       "4  1000565:000005  1000565   1000565:000005  F5R745  METUNv1_00036  \\N   \n",
       "\n",
       "  Transposase of ISThsp18, IS1182 family  \n",
       "0                            Transposase  \n",
       "1                   hypothetical protein  \n",
       "2                            Transposase  \n",
       "3                                 PPIase  \n",
       "4                      Endopeptidase Clp  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "odb_genes = pd.read_csv(\"./../odb9v1_genes.tab\", sep='\\t')\n",
    "odb_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_file.to_csv(\"orthoDB.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Target</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_1</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>64</td>\n",
       "      <td>286</td>\n",
       "      <td>1.400000e-09</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58695</td>\n",
       "      <td>PPA10102</td>\n",
       "      <td>PPA10102 477 554 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>46</td>\n",
       "      <td>238</td>\n",
       "      <td>3.800000e-11</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58696</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "      <td>G4UTF3_NEUT9 256 317 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>936</td>\n",
       "      <td>1635</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58697</td>\n",
       "      <td>ENSFALP00000015835</td>\n",
       "      <td>ENSFALP00000015835 342 597 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>104</td>\n",
       "      <td>989</td>\n",
       "      <td>5.000000e-29</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58698</td>\n",
       "      <td>G1X5W6_ARTOA</td>\n",
       "      <td>G1X5W6_ARTOA 879 1166 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_10001</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>207</td>\n",
       "      <td>456</td>\n",
       "      <td>3.300000e-16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58699</td>\n",
       "      <td>C1GUP1_PARBA</td>\n",
       "      <td>C1GUP1_PARBA 8 91 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              seqid source                         type  start   end  \\\n",
       "0      Transcript_1   LAST  translated_nucleotide_match     64   286   \n",
       "1    Transcript_100   LAST  translated_nucleotide_match     46   238   \n",
       "2   Transcript_1000   LAST  translated_nucleotide_match    936  1635   \n",
       "3  Transcript_10000   LAST  translated_nucleotide_match    104   989   \n",
       "4  Transcript_10001   LAST  translated_nucleotide_match    207   456   \n",
       "\n",
       "          score strand  phase              ID                Name  \\\n",
       "0  1.400000e-09      +    NaN  homology:58695            PPA10102   \n",
       "1  3.800000e-11      -    NaN  homology:58696        G4UTF3_NEUT9   \n",
       "2  2.800000e-31      -    NaN  homology:58697  ENSFALP00000015835   \n",
       "3  5.000000e-29      +    NaN  homology:58698        G1X5W6_ARTOA   \n",
       "4  3.300000e-16      -    NaN  homology:58699        C1GUP1_PARBA   \n",
       "\n",
       "                         Target database  \n",
       "0            PPA10102 477 554 +  OrthoDB  \n",
       "1        G4UTF3_NEUT9 256 317 +  OrthoDB  \n",
       "2  ENSFALP00000015835 342 597 +  OrthoDB  \n",
       "3       G1X5W6_ARTOA 879 1166 +  OrthoDB  \n",
       "4           C1GUP1_PARBA 8 91 +  OrthoDB  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')#[['seqid', 'Name']]\n",
    "new_file = names.dropna(axis=0,how='all')\n",
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.dammit.gff3\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Target</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_1</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>64</td>\n",
       "      <td>286</td>\n",
       "      <td>1.400000e-09</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58695</td>\n",
       "      <td>PPA10102</td>\n",
       "      <td>PPA10102 477 554 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>46</td>\n",
       "      <td>238</td>\n",
       "      <td>3.800000e-11</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58696</td>\n",
       "      <td>G4UTF3_NEUT9</td>\n",
       "      <td>G4UTF3_NEUT9 256 317 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>936</td>\n",
       "      <td>1635</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58697</td>\n",
       "      <td>ENSFALP00000015835</td>\n",
       "      <td>ENSFALP00000015835 342 597 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>104</td>\n",
       "      <td>989</td>\n",
       "      <td>5.000000e-29</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58698</td>\n",
       "      <td>G1X5W6_ARTOA</td>\n",
       "      <td>G1X5W6_ARTOA 879 1166 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_10001</td>\n",
       "      <td>LAST</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>207</td>\n",
       "      <td>456</td>\n",
       "      <td>3.300000e-16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:58699</td>\n",
       "      <td>C1GUP1_PARBA</td>\n",
       "      <td>C1GUP1_PARBA 8 91 +</td>\n",
       "      <td>OrthoDB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              seqid source                         type  start   end  \\\n",
       "0      Transcript_1   LAST  translated_nucleotide_match     64   286   \n",
       "1    Transcript_100   LAST  translated_nucleotide_match     46   238   \n",
       "2   Transcript_1000   LAST  translated_nucleotide_match    936  1635   \n",
       "3  Transcript_10000   LAST  translated_nucleotide_match    104   989   \n",
       "4  Transcript_10001   LAST  translated_nucleotide_match    207   456   \n",
       "\n",
       "          score strand  phase              ID                Name  \\\n",
       "0  1.400000e-09      +    NaN  homology:58695            PPA10102   \n",
       "1  3.800000e-11      -    NaN  homology:58696        G4UTF3_NEUT9   \n",
       "2  2.800000e-31      -    NaN  homology:58697  ENSFALP00000015835   \n",
       "3  5.000000e-29      +    NaN  homology:58698        G1X5W6_ARTOA   \n",
       "4  3.300000e-16      -    NaN  homology:58699        C1GUP1_PARBA   \n",
       "\n",
       "                         Target database  \n",
       "0            PPA10102 477 554 +  OrthoDB  \n",
       "1        G4UTF3_NEUT9 256 317 +  OrthoDB  \n",
       "2  ENSFALP00000015835 342 597 +  OrthoDB  \n",
       "3       G1X5W6_ARTOA 879 1166 +  OrthoDB  \n",
       "4           C1GUP1_PARBA 8 91 +  OrthoDB  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')#[['seqid', 'Name']]\n",
    "new_file = names.dropna(axis=0,how='all')\n",
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Target</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42395</th>\n",
       "      <td>Transcript_9979</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>27</td>\n",
       "      <td>291</td>\n",
       "      <td>9.600000e-66</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:42395</td>\n",
       "      <td>jgi|Emihu1|449249|estExtDG_Genemark1.C_140358</td>\n",
       "      <td>jgi|Emihu1|449249|estExtDG_Genemark1.C_140358 ...</td>\n",
       "      <td>Emihu1_reduced_proteins.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39228</th>\n",
       "      <td>Transcript_9986</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>1</td>\n",
       "      <td>77</td>\n",
       "      <td>1.500000e-31</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:39228</td>\n",
       "      <td>jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038</td>\n",
       "      <td>jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038...</td>\n",
       "      <td>Emihu1_reduced_proteins.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42396</th>\n",
       "      <td>Transcript_9987</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>97</td>\n",
       "      <td>361</td>\n",
       "      <td>7.400000e-64</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:42396</td>\n",
       "      <td>jgi|Emihu1|103643|fgeneshEH_pg.52__22</td>\n",
       "      <td>jgi|Emihu1|103643|fgeneshEH_pg.52__22 173 434 +</td>\n",
       "      <td>Emihu1_reduced_proteins.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10023</th>\n",
       "      <td>Transcript_999</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>494</td>\n",
       "      <td>542</td>\n",
       "      <td>1.800000e-19</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:10023</td>\n",
       "      <td>jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2</td>\n",
       "      <td>jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2 166...</td>\n",
       "      <td>Emihu1_reduced_proteins.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34200</th>\n",
       "      <td>Transcript_9993</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>6.700000e-28</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:34200</td>\n",
       "      <td>jgi|Emihu1|243566|gm1.39200006</td>\n",
       "      <td>jgi|Emihu1|243566|gm1.39200006 78 142 +</td>\n",
       "      <td>Emihu1_reduced_proteins.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seqid        source                              type  start  \\\n",
       "42395  Transcript_9979  shmlast.LAST  conditional_reciprocal_best_LAST     27   \n",
       "39228  Transcript_9986  shmlast.LAST  conditional_reciprocal_best_LAST      1   \n",
       "42396  Transcript_9987  shmlast.LAST  conditional_reciprocal_best_LAST     97   \n",
       "10023   Transcript_999  shmlast.LAST  conditional_reciprocal_best_LAST    494   \n",
       "34200  Transcript_9993  shmlast.LAST  conditional_reciprocal_best_LAST      6   \n",
       "\n",
       "       end         score strand  phase              ID  \\\n",
       "42395  291  9.600000e-66      +    NaN  homology:42395   \n",
       "39228   77  1.500000e-31      +    NaN  homology:39228   \n",
       "42396  361  7.400000e-64      +    NaN  homology:42396   \n",
       "10023  542  1.800000e-19      +    NaN  homology:10023   \n",
       "34200   70  6.700000e-28      +    NaN  homology:34200   \n",
       "\n",
       "                                                 Name  \\\n",
       "42395   jgi|Emihu1|449249|estExtDG_Genemark1.C_140358   \n",
       "39228  jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038   \n",
       "42396           jgi|Emihu1|103643|fgeneshEH_pg.52__22   \n",
       "10023      jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2   \n",
       "34200                  jgi|Emihu1|243566|gm1.39200006   \n",
       "\n",
       "                                                  Target  \\\n",
       "42395  jgi|Emihu1|449249|estExtDG_Genemark1.C_140358 ...   \n",
       "39228  jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038...   \n",
       "42396    jgi|Emihu1|103643|fgeneshEH_pg.52__22 173 434 +   \n",
       "10023  jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2 166...   \n",
       "34200            jgi|Emihu1|243566|gm1.39200006 78 142 +   \n",
       "\n",
       "                            database  \n",
       "42395  Emihu1_reduced_proteins.fasta  \n",
       "39228  Emihu1_reduced_proteins.fasta  \n",
       "42396  Emihu1_reduced_proteins.fasta  \n",
       "10023  Emihu1_reduced_proteins.fasta  \n",
       "34200  Emihu1_reduced_proteins.fasta  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.x.Emihu1_reduced_proteins.fasta.crbl.gff3\"\n",
    "annotations = GFF3Parser(filename=gff_file).read()\n",
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')#[['seqid', 'Name']]\n",
    "new_file = names.dropna(axis=0,how='all')\n",
    "new_file.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42395</th>\n",
       "      <td>Transcript_9979</td>\n",
       "      <td>jgi|Emihu1|449249|estExtDG_Genemark1.C_140358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39228</th>\n",
       "      <td>Transcript_9986</td>\n",
       "      <td>jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42396</th>\n",
       "      <td>Transcript_9987</td>\n",
       "      <td>jgi|Emihu1|103643|fgeneshEH_pg.52__22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10023</th>\n",
       "      <td>Transcript_999</td>\n",
       "      <td>jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34200</th>\n",
       "      <td>Transcript_9993</td>\n",
       "      <td>jgi|Emihu1|243566|gm1.39200006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seqid                                            Name\n",
       "42395  Transcript_9979   jgi|Emihu1|449249|estExtDG_Genemark1.C_140358\n",
       "39228  Transcript_9986  jgi|Emihu1|457020|estExtDG_Genemark1.C_2130038\n",
       "42396  Transcript_9987           jgi|Emihu1|103643|fgeneshEH_pg.52__22\n",
       "10023   Transcript_999      jgi|Emihu1|314592|fgenesh_newKGs_pm.188__2\n",
       "34200  Transcript_9993                  jgi|Emihu1|243566|gm1.39200006"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.x.Emihu1_reduced_proteins.fasta.crbl.gff3\"\n",
    "annotations = GFF3Parser(filename=gff_file).read()\n",
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name']]\n",
    "new_file = names.dropna(axis=0,how='all')\n",
    "new_file.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>protein_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>jgi|Emihu1|198495|gm1.800250</td>\n",
       "      <td>198495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25047</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>jgi|Emihu1|117245|fgeneshEH_pg.330__1</td>\n",
       "      <td>117245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Transcript_1001</td>\n",
       "      <td>jgi|Emihu1|244250|gm1.41400013</td>\n",
       "      <td>244250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3045</th>\n",
       "      <td>Transcript_10010</td>\n",
       "      <td>jgi|Emihu1|99708|fgeneshEH_pg.20__84</td>\n",
       "      <td>99708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Transcript_10012</td>\n",
       "      <td>jgi|Emihu1|254305|gm1.226300001</td>\n",
       "      <td>254305</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  seqid                                   Name protein_id\n",
       "3         Transcript_10           jgi|Emihu1|198495|gm1.800250     198495\n",
       "25047   Transcript_1000  jgi|Emihu1|117245|fgeneshEH_pg.330__1     117245\n",
       "6       Transcript_1001         jgi|Emihu1|244250|gm1.41400013     244250\n",
       "3045   Transcript_10010   jgi|Emihu1|99708|fgeneshEH_pg.20__84      99708\n",
       "8      Transcript_10012        jgi|Emihu1|254305|gm1.226300001     254305"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "new_file['protein_id']=new_file.Name.str.split(\"|\",expand=True,)[2]\n",
    "new_file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_file.to_csv(\"ehux_protein_annotations.csv\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/brisbin/miniconda2/envs/py3.dammit/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>score</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_11149</td>\n",
       "      <td>9.900000e-10</td>\n",
       "      <td>NP_045875.1 septum site-determining protein (c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Transcript_16167</td>\n",
       "      <td>1.300000e-08</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Transcript_18487</td>\n",
       "      <td>7.800000e-138</td>\n",
       "      <td>XP_001713182.1 cell division protein FtsZ [Gui...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Transcript_18490</td>\n",
       "      <td>1.200000e-09</td>\n",
       "      <td>lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Transcript_18526</td>\n",
       "      <td>1.100000e-08</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Transcript_18919</td>\n",
       "      <td>1.600000e-08</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Transcript_20017</td>\n",
       "      <td>6.500000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Transcript_24313</td>\n",
       "      <td>1.100000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Transcript_26127</td>\n",
       "      <td>5.200000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_26867</td>\n",
       "      <td>1.600000e-08</td>\n",
       "      <td>lcl|NC_010144.1_prot_XP_005538591.1_1 [locus_t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_27723</td>\n",
       "      <td>9.200000e-132</td>\n",
       "      <td>XP_001713182.1 cell division protein FtsZ [Gui...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transcript_27837</td>\n",
       "      <td>8.000000e-10</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Transcript_29318</td>\n",
       "      <td>2.000000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Transcript_29767</td>\n",
       "      <td>8.100000e-09</td>\n",
       "      <td>XP_001702662.1 dynamin-related GTPase [Chlamyd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_31610</td>\n",
       "      <td>3.800000e-69</td>\n",
       "      <td>BAC55068.1 dynamin related protein involved in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Transcript_32317</td>\n",
       "      <td>8.600000e-14</td>\n",
       "      <td>lcl|AB673434.1_prot_BAM11092.1_1 [gene=drp5B] ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Transcript_3862</td>\n",
       "      <td>2.100000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Transcript_4145</td>\n",
       "      <td>1.200000e-08</td>\n",
       "      <td>XP_001700508.1 plastid division protein FtsZ2 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Transcript_9042</td>\n",
       "      <td>1.100000e-11</td>\n",
       "      <td>lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Transcript_971</td>\n",
       "      <td>3.500000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               seqid          score  \\\n",
       "2   Transcript_11149   9.900000e-10   \n",
       "19  Transcript_16167   1.300000e-08   \n",
       "5   Transcript_18487  7.800000e-138   \n",
       "8   Transcript_18490   1.200000e-09   \n",
       "11  Transcript_18526   1.100000e-08   \n",
       "13  Transcript_18919   1.600000e-08   \n",
       "16  Transcript_20017   6.500000e-09   \n",
       "17  Transcript_24313   1.100000e-09   \n",
       "15  Transcript_26127   5.200000e-09   \n",
       "0   Transcript_26867   1.600000e-08   \n",
       "3   Transcript_27723  9.200000e-132   \n",
       "18  Transcript_27837   8.000000e-10   \n",
       "14  Transcript_29318   2.000000e-09   \n",
       "21  Transcript_29767   8.100000e-09   \n",
       "1   Transcript_31610   3.800000e-69   \n",
       "20  Transcript_32317   8.600000e-14   \n",
       "12   Transcript_3862   2.100000e-09   \n",
       "22   Transcript_4145   1.200000e-08   \n",
       "9    Transcript_9042   1.100000e-11   \n",
       "10    Transcript_971   3.500000e-09   \n",
       "\n",
       "                                                 Name  \n",
       "2   NP_045875.1 septum site-determining protein (c...  \n",
       "19  XP_001702420.1 plastid division protein [Chlam...  \n",
       "5   XP_001713182.1 cell division protein FtsZ [Gui...  \n",
       "8   lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...  \n",
       "11  XP_001702420.1 plastid division protein [Chlam...  \n",
       "13  XP_001702420.1 plastid division protein [Chlam...  \n",
       "16  XP_001702420.1 plastid division protein [Chlam...  \n",
       "17  XP_001702420.1 plastid division protein [Chlam...  \n",
       "15  XP_001702420.1 plastid division protein [Chlam...  \n",
       "0   lcl|NC_010144.1_prot_XP_005538591.1_1 [locus_t...  \n",
       "3   XP_001713182.1 cell division protein FtsZ [Gui...  \n",
       "18  XP_001702420.1 plastid division protein [Chlam...  \n",
       "14  XP_001702420.1 plastid division protein [Chlam...  \n",
       "21  XP_001702662.1 dynamin-related GTPase [Chlamyd...  \n",
       "1   BAC55068.1 dynamin related protein involved in...  \n",
       "20  lcl|AB673434.1_prot_BAM11092.1_1 [gene=drp5B] ...  \n",
       "12  XP_001702420.1 plastid division protein [Chlam...  \n",
       "22  XP_001700508.1 plastid division protein FtsZ2 ...  \n",
       "9   lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...  \n",
       "10  XP_001702420.1 plastid division protein [Chlam...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_file = \"pc_euk_seqs.fasta.x.chloroplast_division_proteins.fasta.crbl.gff3\"\n",
    "annotations = GFF3Parser(filename=gff_file).read()\n",
    "names = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid','score', 'Name']]\n",
    "new_file = names.dropna(axis=0,how='all')\n",
    "new_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "namemap= pd.read_csv(\"pc_euk_seqs.fasta.dammit.namemap.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "#new = namemap['original'].str.split(\" \", n = 1, expand = True) \n",
    "#new.head()\n",
    "namemap['original'] = new[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>original</th>\n",
       "      <th>renamed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN29218_c0_g1_i1</td>\n",
       "      <td>Transcript_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN29280_c0_g1_i1</td>\n",
       "      <td>Transcript_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN29255_c0_g1_i1</td>\n",
       "      <td>Transcript_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN29291_c0_g1_i1</td>\n",
       "      <td>Transcript_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN29211_c0_g1_i3</td>\n",
       "      <td>Transcript_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   original       renamed\n",
       "0  TRINITY_DN29218_c0_g1_i1  Transcript_0\n",
       "1  TRINITY_DN29280_c0_g1_i1  Transcript_1\n",
       "2  TRINITY_DN29255_c0_g1_i1  Transcript_2\n",
       "3  TRINITY_DN29291_c0_g1_i1  Transcript_3\n",
       "4  TRINITY_DN29211_c0_g1_i3  Transcript_4"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "namemap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>trinity</th>\n",
       "      <th>seqid</th>\n",
       "      <th>score</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN5355_c1_g1_i1</td>\n",
       "      <td>Transcript_971</td>\n",
       "      <td>3.500000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN874_c0_g3_i1</td>\n",
       "      <td>Transcript_3862</td>\n",
       "      <td>2.100000e-09</td>\n",
       "      <td>XP_001702420.1 plastid division protein [Chlam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN4417_c0_g2_i1</td>\n",
       "      <td>Transcript_4145</td>\n",
       "      <td>1.200000e-08</td>\n",
       "      <td>XP_001700508.1 plastid division protein FtsZ2 ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN1241_c0_g2_i2</td>\n",
       "      <td>Transcript_9042</td>\n",
       "      <td>1.100000e-11</td>\n",
       "      <td>lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN8918_c0_g1_i4</td>\n",
       "      <td>Transcript_11149</td>\n",
       "      <td>9.900000e-10</td>\n",
       "      <td>NP_045875.1 septum site-determining protein (c...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   trinity             seqid         score  \\\n",
       "0  TRINITY_DN5355_c1_g1_i1    Transcript_971  3.500000e-09   \n",
       "1   TRINITY_DN874_c0_g3_i1   Transcript_3862  2.100000e-09   \n",
       "2  TRINITY_DN4417_c0_g2_i1   Transcript_4145  1.200000e-08   \n",
       "3  TRINITY_DN1241_c0_g2_i2   Transcript_9042  1.100000e-11   \n",
       "4  TRINITY_DN8918_c0_g1_i4  Transcript_11149  9.900000e-10   \n",
       "\n",
       "                                                Name  \n",
       "0  XP_001702420.1 plastid division protein [Chlam...  \n",
       "1  XP_001702420.1 plastid division protein [Chlam...  \n",
       "2  XP_001700508.1 plastid division protein FtsZ2 ...  \n",
       "3  lcl|AB673433.1_prot_BAM11091.1_1 [gene=ftsZ] [...  \n",
       "4  NP_045875.1 septum site-determining protein (c...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "namemap.columns = [\"trinity\", \"seqid\"]\n",
    "\n",
    "chlordiv = pd.merge(namemap, new_file, on = \"seqid\", how = \"inner\" )\n",
    "\n",
    "chlordiv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "chlordiv.to_csv(\"chlordiv_annotations.csv\", index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
